src/graphql/resolvers.js

```diff
@@ -174,6 +174,43 @@ module.exports = function Resolvers (ssb) {
     })
   }
 
+  /**
+   * Gets all the public threads that room members have posted or replied to
+   * @param {object} opts - optional parameters
+   * @param {int} [opts.limit=10] - max amount of threads to return
+   * @param {int} opts.threadMaxSize - max amount of messages in each thread to return
+   */
+  const getPublicThreads = (opts) => {
+    const { threadMaxSize, limit = 10 } = opts
+    const members = ssb.room.members()
+  
+    return new Promise((resolve, reject) => {
+      pull(
+        ssb.threads.public({ threadMaxSize, allowlist: ['post'], following: true }),
+        // hacky! Only threads that contain a message from a member of the room
+        // this is because the room is following some users who are not members
+        pull.filter(thread => {
+          return thread.messages.some(message => {
+            return members.includes(message.value?.author)
+          })
+        }),
+        pull.take(limit),
+        pull.collect((err, threads) => {
+          if (err) return reject(err)
+
+          const res = threads.map(({ messages }) => {
+            return {
+              id: messages[0].key,
+              messages
+            }
+          })
```

Comment:
only question I have is "does this omit messages from people who don't want to be publiclly web hosted?"